ddtrace/contrib/psycopg/patch.py

```diff
@@ -39,7 +39,8 @@
     # using unwrap
     _original_connect = psycopg_import.Connection.__dict__["connect"]
     _original_async_connect = psycopg_import.AsyncConnection.__dict__["connect"]
-except ImportError:
+# AttributeError can happen due to circular imports under certain integration methods
+except (ImportError, AttributeError):
```

Comment:
Instead of catching circular imports here would it be possible to update ModuleWatchDog so that it detects and tries reconciles circular imports while patching integrations? 

Resolving this issue in all integrations could become a painful game of whack a mole

Reply:
Great idea.